<?xml version="1.0"?>
<launch>
    <!-- video_stream_provider has to be an index or an array of indices of video devices -->
    <!-- for example 0 or [0, 2, 4, 6, 8] -->
    <arg name="video_stream_provider" default="0" />
    <!-- frames per second for publish -->
    <arg name="fps" default="30" />
    <!-- frame_id for the camera -->
    <arg name="frame_id" default="world" />
    <!-- defines if the script should try to automatically reconnect if the frame was not captured -->
    <arg name="retry_on_fail" default="false" />
    <!-- path to the connector chart if should be used -->
    <arg name="camera_connector_chart" default="" />
    <arg name="buffer_queue_size" default="1" />
    <arg name="python_node" default="false" />

    <node pkg="multicam_server" type="start_streamers.py" name="camera_stream" output="screen">
        <param name="video_stream_provider" type="string" value="$(arg video_stream_provider)" />
        <param name="fps" type="double" value="$(arg fps)" />
        <param name="frame_id" type="string" value="$(arg frame_id)" />
        <param name="retry_on_fail" type="bool" value="$(arg retry_on_fail)" />
        <param name="camera_connector_chart" type="string" value="$(arg camera_connector_chart)" />
        <param name="buffer_queue_size" type="int" value="$(arg buffer_queue_size)" />
        <param name="python_node" type="bool" value="$(arg python_node)" />
    </node>
</launch>
